providers/resteasy-validator-provider-11/src/main/java/org/jboss/resteasy/plugins/validation/GeneralValidatorImpl.java

```
@@ -416,9 +419,24 @@ static protected String convertArrayToString(Object o)
     * Here, the "super" relationship is reflexive.  That is, a method
     * is a super method of itself.
     */
-   protected Method getSuperMethod(Method method, Class<?> clazz)
+   protected Method getSuperMethod(Method method, final Class<?> clazz)
    {
-      Method[] methods = clazz.getDeclaredMethods();
+      Method[] methods = new Method[0];
+      try {
+         if (System.getSecurityManager() == null) {
+            methods = clazz.getDeclaredMethods();
+         } else {
+            methods = AccessController.doPrivileged(new PrivilegedExceptionAction<Method[]>() {
+               @Override
+               public Method[] run() throws Exception {
+                  return clazz.getDeclaredMethods();
+               }
+            });
+         }
+      } catch (PrivilegedActionException pae) {
+         LogMessages.LOGGER.error(pae.getMessage());

```

I think it should throw exception instead of logging it.